### ISSUE TITLE
IMDB - Search Result Change

### ISSUE THREAD

**Issue by BBen1979** [FabianBeiner/PHP-IMDB-Grabber#177]
Hi there,

Some issues with returned search URL, been a few months now since they updated IMDB.

EG:
https://www.imdb.com/title/tt10738906/?ref_=fn_tt_tt_1 end up at https://www.imdb.com/find?q=The+Feast+(2021)&s=tt
https://www.imdb.com/title/tt14369780/?ref_=fn_tt_tt_1 end up at https://www.imdb.com/find?q=Lady+Chatterleys+Lover+(2022)&s=tt
https://www.imdb.com/title/tt19875906/?ref_=fn_tt_tt_1 end up at https://www.imdb.com/find?q=A+Wonderful+Time+of+the+Year+(2022)&s=tt

Basically pretty much every result end up on the search page instead of the good link.

EG:

---

```
<?php include_once 'imdb.class.php'; $IMDB = new IMDB('The Feast 2021'); if ($IMDB->isReady) {
    print_r($IMDB->getAll());
} else {
    echo 'Movie not found. 😞';
}

?>

```

URL: Will end up as Url [value] => https://www.imdb.com/find?q=The+Feast+2021&s=tt )

---

Thanks

**Comment by bla0r** [comment]
Will taking a look into it.

**Comment by FabianBeiner** [comment]
I can't reproduce any of this, @BBen1979. :( Let us use one of @Artbabe's examples:

Searching for “Dinner 2016” gives us these results at IMDb directly (https://www.imdb.com/find?q=Dinner+2016&ref_=nv_sr_sm): 

![image](https://user-images.githubusercontent.com/86269/206027386-92ecd59a-1541-4607-9393-e4258749b3fa.png)

Using the latest version of the script returns:

![image](https://user-images.githubusercontent.com/86269/206027791-39f20fd8-fab4-4f32-9a87-1cce0614a3f5.png)

If we force the script to search only for movies (```$oIMDB = new IMDB('Dinner 2016', null, 'movie');```) the result changes too:

![image](https://user-images.githubusercontent.com/86269/206029003-a49a65f0-6d80-4924-b3f8-857941959bb6.png)

And so does the script:

![image](https://user-images.githubusercontent.com/86269/206029132-55e4409e-87d3-4765-9f0d-ab0dd3aa5f81.png)

So, as you see, it behaves exactly as supposed to. 

If I'm missing something here, please enable debug, run it again and post the output. Thanks.

**Comment by bla0r** [comment]
Can confirm it. Please provide some debug notes.

**Comment by BBen1979** [comment]
Updated to the latest imdb.class.php fixed my issue. Maybe i had a older version with the bug I posted.
Sorry about that, and thank you guys for all the support and great work. Love this script :P

**Comment by Artbabe** [comment]
hello
sorry , i didnt see it
@bla0r : you can go there and see if your scripts is working or not 

just a note
I tested your current script which is very different from old one

Nanny 2022

test with your current script
poster didnt work also picked wrong movie as well

also
please show me a place where I can upload my previous script
because I changed it several times

my script has 2 problems
1- [the titles I mentioned in earlier post of mine , those which contain character which I added a lot of exception in that array you created [that & vs and ... one]

and
when there are 2 numbers in title , [2 four digits] it almost always gives wrong result]

where I can upload my script?

**Comment by Artbabe** [comment]
OK
sorry
I think I posted in wrong place

thanks for everything [both of you , Maestro Fabian and also BL who helped us so much ]

have a nice day

**Comment by bla0r** [comment]
@Artbabe if it's an issue please post a new report.

**Comment by Artbabe** [comment]
@bla0r :
I already told you in that other thread
Search of imdb changed
the search pattern must be updated

Tides 2017
Maze 2017
200 M.P.H 2011
Isolated 2022

these are last 30 minutes' items which took wrong one
also I dont use this script for tv anymore , tv of imdb isnt good
just movie

please show me a place and let me upload my own script which is better than all ,you changed the whole script many times until you created this one which I updated in last months and debugged [using your tips]

before the day I reported the change in imdb's search  [few weeks ago] my script had/has only 1 problem, which is
when there are two 4 digits numbers , it considers 2 years and misses almost 100%

but now , we must fix many posts manually due to this Search's issue

if you are in rush if you just post a comment in site I can answer faster please mention just your id 
if not I will check this tread in few hours

**Comment by Artbabe** [comment]
BTW
if you have any solution which I can forward to imdb team , I would be grateful
do you have any solution for their recent search problem?

Mayor 2020

Among Us 2019

I am in touch with one of their team

**Comment by bla0r** [comment]
Can confirm the problems of your given titles. I will take a deeper look into it.

**Comment by Artbabe** [comment]
in the meanwhile
no thread/post should be without maestro's name
Fabian
who shared his amazing skills with us , years ago 
just wanted to pay my respect to him ,again

**Comment by Artbabe** [comment]
Spider 2002
is funny case which just happened

@bla0r 
I know exactly how good you are
but let me be honest , I dont think you can solve this search's bug [I call it]

g-d knows what imdb team has done last month
they literally destroyed the search function

**Comment by Artbabe** [comment]
mistery solved

weirdly !!
a coder solved this problem

please do this way: [add &exact=true ]

https://www.imdb.com/find/?q=spider%202002**&s=tt&exact=true**&ref_=fn_tt_ex

**Comment by Artbabe** [comment]
I closed that post which you asked me to close

I have a question
where I need to apply this &exact=true ?
can you tell me?

this is my script's biggest problem
I can handle other issues more/less
this is what is causing problem


                $this->sUrl = 'https://www.imdb.com/find?q=' . rawurlencode(str_replace(' ', ' ', $sSearch)) . $sParameters;                


is this where I should apply it?

**Comment by bla0r** [comment]
A new commit with changes coming soon.

**Comment by Artbabe** [comment]
Camping 2 2010

check this one too
kinda nice item
my script gets
https://www.imdb.com/title/tt1252380/

should get
https://www.imdb.com/title/tt1503096/

**Comment by bla0r** [comment]
fixed. Check #181

**Comment by Artbabe** [comment]
ok
i will report the result here
appreciated

**Comment by Artbabe** [comment]
Bl:
thank you for everything
I am getting syntax error when I tried to use it

Fatal error: Call to undefined function array_column() in /imdb.class.php on line 450

which is
                $matches = array_column($rData, 'match');

can you confirm if it is because of my changes or you also get it?

**Comment by bla0r** [comment]
Update your php.

**Comment by Artbabe** [comment]
oh uh
its not possible

ok 
so this is the reason
then I am dead man !

ok man thank you
it didnt work for me I hope it works for others
have a nice year

**Comment by Artbabe** [comment]
Goooood to see you Maestro Fabian

my php is 5.4.16 in that server which I use for test
last update = end of 2020 as far as I remember

this script is not doing what I expect [poster+synopsis , AFTER detecting the ID]
and since yesterday I am getting syntax error so I cant test the latest script

please close this thread
unfortunately my journey is over with this cool masterpiece

**Comment by FabianBeiner** [comment]
> my php is 5.4.16 in that server which I use for test last update = end of 2020 as far as I remember

[PHP 5.4.45 was published on 06 Jun 2013](https://www.php.net/ChangeLog-5.php#5.4.16). You are running a server with major securities issues there. And have a look at https://www.php.net/supported-versions.php to see for yourself with PHP versions are actually current and still supported.

But yeah, either get the script downgraded manually by someone or try to find a replacement. 🙏🏻

**Comment by Artbabe** [comment]
Using username "root".
Last login: Tue Jan  3 15:29:22 2023 from **
[root@localhost ~]# php -v
PHP 5.4.16 (cli) (built: Apr  1 2020 04:07:17)

Yes , I am woking to solve the search problem

again: thank you [both] for all of these years

**Comment by bla0r** [comment]
built means compiled... you can update your php version and you'll be fine.

**Comment by Artbabe** [comment]
hello BL:
Updating php is not option for me because I dont have enough human-resource to rebuild all of my platform
platform writen in 2013 , updated few times and last update was 2020

I am Reserve IDF Medic and my contract ends at 2024 December
bcuz I must stay Human!! and wont break under pressure of WAR ,I must continue this hobby for 23 more months

so ,doesnt worth to rebuild everything!

and yes
I will find something to add Exact=true to old script
even a simple title like
Merrick 2017
miisses by my current script

the guy who wrote my platform , doesnt even answer my call [after he found out I am IDF medic ! , just medic !!]

**Comment by FabianBeiner** [comment]
> are you working on a new version? that sounds good 👍

Not actively. ;) But I would do so much differently codewise nowadays, that I once in a while think about redoing it. On the other hand: I never ever used that script myself. 🤷🏻‍♂️ So there is no pressure behind this for me.

**Comment by bla0r** [comment]
> Merrick 2017

works at current script. Ask your coder what to do. You can replace array_column with array_map and an extended function.

cheers

**Comment by Artbabe** [comment]
@bla0r 
its not that 
since the last month , I cant use this script
for me this project is over 
I asked people to help me ,I hope I can find solution
thanks for all of these years @FabianBeiner   and thanks to you @bla0r  for fixing it many times

**Comment by Artbabe** [comment]
hello
imdb's one of admins checked the case

but still , without exact=true , we have errors or wrong picked id

I asked them to set exact=true , default
and they sorta did it [not exactly]
but we still need to use exact=true for some items/day

which I still cant integrate to my script [I integrated exact=true but didnt change result at all , weird!!]
i think I am using bl's revsion which that [percent] also is causing this case

I just wanted you be aware of these stuff

I just hope I can convince them to fix search's problem

NOTE:
AFTER A WHILE , BECAUSE OF A SMART SCRIPT IN IMDB , IT STARTS PICKING RIGHT ITEM
but it takes 1 to 2 weeks to do it

this smart script monitors user's behaviours
for example
when they see my site sends 166k daily tier1 to a wrong item and then user chooses right one , it kinda brings the case up [in seearch]
but
after 1-2 weeks
and this is not good for me !!
but its good for imdb

it seems I must try , try ... to finally get proper script which can solve my problems

have a nice day , both of you amazing coders
Dear BL who solved my problem once , a year ago
and , "THE"  master Fabian

**Comment by Artbabe** [comment]
can you show me a person who used your updated php version and can confirm that it gets the imdb movie correctly?
i mean regarding exact true's matter
then I can ask someone to turn to the php version which sites also can use it
whoever I know and I recommended them to use Dear Fabian's updated script , say ; they cant use it [due to different reasons but mostly bcuz when they try , it gives blank response [I assume its php?!]

we technically got crippled since 2 months ago
rate of error is insanely high
thanks G-d I only use for movie also only synopsis/poster [after the id recognizes]

maybe among users someone can give us the advice which can end this problem
imdb's team also cant change their search's code due to lack of resource [unbelievable but true !]

I want it always uses exact=true
which I hope it reduces the error to acceptable range

**Comment by Artbabe** [comment]
all problems solved

thanks guys
